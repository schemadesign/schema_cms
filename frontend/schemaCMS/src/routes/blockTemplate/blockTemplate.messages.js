/* eslint-disable max-len */
import { defineMessages } from 'react-intl';

export default defineMessages({
  title: {
    id: 'blockTemplate.title',
    defaultMessage: 'Block Template',
  },
  subtitle: {
    id: 'blockTemplate.subtitle',
    defaultMessage: 'Edit',
  },
  cancel: {
    id: 'blockTemplate.cancel',
    defaultMessage: 'Back',
  },
  save: {
    id: 'blockTemplate.save',
    defaultMessage: 'Save',
  },
  cancelRemoval: {
    id: 'blockTemplate.cancelRemoval',
    defaultMessage: 'Cancel',
  },
  confirmRemoval: {
    id: 'blockTemplate.confirmRemoval',
    defaultMessage: 'Confirm',
  },
  removeTitle: {
    id: 'blockTemplate.removeTitle',
    defaultMessage: 'Are you sure you want to remove this block template?',
  },
});
