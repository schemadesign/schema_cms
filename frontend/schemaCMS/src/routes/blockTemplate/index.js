export { default as BlockTemplate } from './blockTemplate.container';
