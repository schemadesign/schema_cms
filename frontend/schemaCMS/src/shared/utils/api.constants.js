const API_PREFIX = process.env.REACT_APP_BASE_API_URL;

export const AUTH_PATH = `${API_PREFIX}/auth/login/auth0`;
export const TOKEN_PATH = '/auth/token/';
export const USER_PATH = '/user/';
