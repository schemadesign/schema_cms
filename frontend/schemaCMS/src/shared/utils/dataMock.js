export const tableData = [
  {
    'First Name': 'George',
    'Last Name': 'Washington',
    'Full Name': 'George Washington',
    Birthplace: 'Westmoreland County',
    'State of Birth': 'VA',
    Birthday: '02/22/1732',
    Url: 'some.url.com',
  },
  {
    'First Name': 'John',
    'Last Name': 'Adams',
    'Full Name': 'John Adams',
    Birthplace: 'Braintree',
    'State of Birth': 'MA',
    Birthday: '10/30/1735',
    Url: 'some.url.com',
  },
  {
    'First Name': 'Thomas',
    'Last Name': 'Jefferson',
    'Full Name': 'Thomas Jefferson',
    Birthplace: 'Shadwell',
    'State of Birth': 'VA',
    Birthday: '04/13/1743',
    Url: 'some.url.com',
  },
  {
    'First Name': 'James',
    'Last Name': 'Madison',
    'Full Name': 'James Madison',
    Birthplace: 'Port Conway',
    'State of Birth': 'VA',
    Birthday: '03/16/1751',
    Url: 'some.url.com',
  },
  {
    'First Name': 'James',
    'Last Name': 'Monroe',
    'Full Name': 'James Monroe',
    Birthplace: 'Monroe Hall',
    'State of Birth': 'VA',
    Birthday: '04/28/1758',
    Url: 'some.url.com',
  },
];

export const tableFields = {
  'First Name': {},
  'Last Name': {},
  'Full Name': {},
  Birthplace: {},
  'State of Birth': {},
  Birthday: {},
  Url: {},
};
