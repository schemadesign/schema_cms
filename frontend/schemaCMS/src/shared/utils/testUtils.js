import { Router } from 'react-router';
import { IntlProvider } from 'react-intl';
import { act, create } from 'react-test-renderer';
import React from 'react';
import { ThemeProvider } from 'styled-components';
import { Theme } from 'schemaUI';
import PropTypes from 'prop-types';

import { DEFAULT_LOCALE, translationMessages } from '../../i18n';
import browserHistory from './history';

const ProvidersWrapper = ({ children }) => (
  <Router history={browserHistory}>
    <IntlProvider locale={DEFAULT_LOCALE} messages={translationMessages[DEFAULT_LOCALE]}>
      <ThemeProvider theme={Theme.dark}>{children}</ThemeProvider>
    </IntlProvider>
  </Router>
);

ProvidersWrapper.propTypes = {
  children: PropTypes.node.isRequired,
};

export const makeContextRenderer = async component => {
  let wrapper = null;

  await act(async () => {
    wrapper = await create(<ProvidersWrapper>{component}</ProvidersWrapper>);
  });

  return wrapper;
};
