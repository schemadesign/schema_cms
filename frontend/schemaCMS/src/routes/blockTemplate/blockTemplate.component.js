import React, { useState, memo, useEffect } from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage, useIntl } from 'react-intl';
import { useFormik } from 'formik';
import Helmet from 'react-helmet';
import { useHistory, useParams } from 'react-router';
import { useEffectOnce } from 'react-use';
import { pick } from 'ramda';

import { Container } from './blockTemplate.styles';
import messages from './blockTemplate.messages';
import { MobileMenu } from '../../shared/components/menu/mobileMenu';
import { errorMessageParser, filterMenuOptions } from '../../shared/utils/helpers';
import {
  BLOCK_TEMPLATES_DELETE_ELEMENTS,
  BLOCK_TEMPLATES_SCHEMA,
  ELEMENT_ID,
  ELEMENT_NAME,
  ELEMENT_PARAMS,
  ELEMENT_TYPE,
  INITIAL_VALUES,
} from '../../modules/blockTemplates/blockTemplates.constants';
import { BlockTemplateForm } from '../../shared/components/blockTemplateForm';
import { BackButton, NavigationContainer, NextButton } from '../../shared/components/navigation';
import { getProjectMenuOptions } from '../project/project.constants';
import { LoadingWrapper } from '../../shared/components/loadingWrapper';
import reportError from '../../shared/utils/reportError';
import { Modal, ModalActions, modalStyles, ModalTitle } from '../../shared/components/modal/modal.styles';
import {
  blockTemplatesMessage,
  libraryMessage,
  ProjectBreadcrumbs,
  projectMessage,
  tabMessage,
  templateMessage,
  templatesMessage,
} from '../../shared/components/projectBreadcrumbs';

const getBreadcrumbsItems = (project, { id, name }) => [
  {
    path: `/project/${project.id}/`,
    span: projectMessage,
    h3: project.title,
  },
  {
    path: `/project/${project.id}/templates`,
    span: tabMessage,
    h3: templatesMessage,
  },
  {
    path: `/project/${project.id}/block-templates`,
    span: libraryMessage,
    h3: blockTemplatesMessage,
  },
  {
    path: `/block-template/${id}`,
    active: true,
    span: templateMessage,
    h3: name,
  },
];

export const BlockTemplate = memo(
  ({
    updateBlockTemplate,
    fetchBlockTemplate,
    removeBlockTemplate,
    copyBlockTemplate,
    fetchProject,
    userRole,
    blockTemplate: { name, elements, isAvailable, allowEdit, project: projectId },
    project,
  }) => {
    const { blockTemplateId } = useParams();
    const history = useHistory();
    const intl = useIntl();
    const [loading, setLoading] = useState(true);
    const [updateLoading, setUpdateLoading] = useState(false);
    const [error, setError] = useState(null);
    const [removeModalOpen, setRemoveModalOpen] = useState(false);
    const [removeLoading, setRemoveLoading] = useState(false);
    const handleConfirmRemove = async () => {
      try {
        setRemoveLoading(true);
        await removeBlockTemplate({ blockTemplateId });
        history.push(`/project/${project.id}/block-templates`);
      } catch (e) {
        reportError(e);
        setRemoveLoading(false);
      }
    };

    const handleBackClick = () => history.push(`/project/${project.id}/block-templates`);

    const menuOptions = getProjectMenuOptions(project.id);
    const { handleSubmit, isValid, dirty, ...restFormikProps } = useFormik({
      initialValues: {
        ...INITIAL_VALUES,
        name,
        isAvailable,
        allowEdit,
        elements: elements.map(element => ({
          ...pick([ELEMENT_NAME, ELEMENT_TYPE, ELEMENT_ID, ELEMENT_PARAMS], element),
          key: element.id,
        })),
        [BLOCK_TEMPLATES_DELETE_ELEMENTS]: [],
      },
      enableReinitialize: true,
      validationSchema: () => BLOCK_TEMPLATES_SCHEMA,
      onSubmit: async (formData, { setErrors }) => {
        try {
          setUpdateLoading(true);
          formData.elements = formData.elements.map((data, index) => ({ ...data, order: index }));
          await updateBlockTemplate({ blockTemplateId, formData });
          setUpdateLoading(false);
        } catch (errors) {
          reportError(errors);
          const { formatMessage } = intl;
          const errorMessages = errorMessageParser({ errors, messages, formatMessage });

          setErrors(errorMessages);
          setUpdateLoading(false);
        }
      },
    });

    useEffectOnce(() => {
      (async () => {
        try {
          await fetchBlockTemplate({ blockTemplateId });
        } catch (e) {
          reportError(e);
          setError(e);
        } finally {
          setLoading(false);
        }
      })();
    });

    useEffect(() => {
      if (projectId) {
        fetchProject({ projectId });
      }
    }, [projectId]);

    const title = <FormattedMessage {...messages.title} />;
    const subtitle = <FormattedMessage {...messages.subtitle} />;

    return (
      <Container>
        <Helmet title={intl.formatMessage(messages.title)} />
        <MobileMenu headerTitle={title} headerSubtitle={subtitle} options={filterMenuOptions(menuOptions, userRole)} />
        <LoadingWrapper loading={loading} error={error}>
          <ProjectBreadcrumbs items={getBreadcrumbsItems(project, { id: blockTemplateId, name })} />
          <form onSubmit={handleSubmit}>
            <BlockTemplateForm
              title={title}
              blockTemplateId={blockTemplateId}
              setRemoveModalOpen={setRemoveModalOpen}
              isValid={isValid}
              dirty={dirty}
              copyBlockTemplate={copyBlockTemplate}
              {...restFormikProps}
            />
            <NavigationContainer fixed>
              <BackButton id="cancelBtn" type="button" onClick={handleBackClick}>
                <FormattedMessage {...messages.cancel} />
              </BackButton>
              <NextButton
                id="createBlockTemplate"
                type="submit"
                loading={updateLoading}
                disabled={!isValid || !dirty || updateLoading}
              >
                <FormattedMessage {...messages.save} />
              </NextButton>
            </NavigationContainer>
          </form>
        </LoadingWrapper>
        <Modal ariaHideApp={false} isOpen={removeModalOpen} contentLabel="Confirm Removal" style={modalStyles}>
          <ModalTitle>
            <FormattedMessage {...messages.removeTitle} />
          </ModalTitle>
          <ModalActions>
            <BackButton onClick={() => setRemoveModalOpen(false)} disabled={removeLoading}>
              <FormattedMessage {...messages.cancelRemoval} />
            </BackButton>
            <NextButton
              id="confirmRemovalBtn"
              onClick={handleConfirmRemove}
              loading={removeLoading}
              disabled={removeLoading}
            >
              <FormattedMessage {...messages.confirmRemoval} />
            </NextButton>
          </ModalActions>
        </Modal>
      </Container>
    );
  }
);

BlockTemplate.propTypes = {
  blockTemplate: PropTypes.object.isRequired,
  project: PropTypes.object.isRequired,
  userRole: PropTypes.string.isRequired,
  fetchBlockTemplate: PropTypes.func.isRequired,
  updateBlockTemplate: PropTypes.func.isRequired,
  removeBlockTemplate: PropTypes.func.isRequired,
  fetchProject: PropTypes.func.isRequired,
  copyBlockTemplate: PropTypes.func.isRequired,
};
