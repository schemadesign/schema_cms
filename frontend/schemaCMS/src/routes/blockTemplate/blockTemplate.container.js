import { connect } from 'react-redux';
import { bindPromiseCreators, promisifyRoutine } from 'redux-saga-routines';
import { createStructuredSelector } from 'reselect';
import { withRouter } from 'react-router-dom';
import { hot } from 'react-hot-loader';
import { compose } from 'ramda';

import { BlockTemplate } from './blockTemplate.component';
import { BlockTemplatesRoutines, selectBlockTemplate, selectBlockTemplates } from '../../modules/blockTemplates';
import { selectUserRole } from '../../modules/userProfile';
import { selectProject } from '../../modules/project';

const mapStateToProps = createStructuredSelector({
  blockTemplate: selectBlockTemplate,
  blockTemplates: selectBlockTemplates,
  userRole: selectUserRole,
  project: selectProject,
});

export const mapDispatchToProps = dispatch => ({
  ...bindPromiseCreators(
    {
      fetchBlockTemplate: promisifyRoutine(BlockTemplatesRoutines.fetchBlockTemplate),
      updateBlockTemplate: promisifyRoutine(BlockTemplatesRoutines.updateBlockTemplate),
      fetchBlockTemplates: promisifyRoutine(BlockTemplatesRoutines.fetchBlockTemplates),
      removeBlockTemplate: promisifyRoutine(BlockTemplatesRoutines.removeBlockTemplate),
    },
    dispatch
  ),
});

export default compose(
  hot(module),
  connect(
    mapStateToProps,
    mapDispatchToProps
  ),
  withRouter
)(BlockTemplate);
