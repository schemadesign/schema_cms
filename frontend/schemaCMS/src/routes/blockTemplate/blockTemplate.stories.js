import React from 'react';
import { storiesOf } from '@storybook/react';

import { BlockTemplate } from './blockTemplate.component';
import { blockTemplate } from '../../modules/blockTemplates/blockTemplates.mocks';
import { ROLES } from '../../modules/userProfile/userProfile.constants';
import { withTheme } from '../../.storybook/decorators';

export const defaultProps = {
  blockTemplate,
  project: {
    id: 'projectId',
  },
  userRole: ROLES.ADMIN,
  fetchBlockTemplate: Function.prototype,
  updateBlockTemplate: Function.prototype,
};

storiesOf('BlockTemplate', module)
  .addDecorator(withTheme())
  .add('Default', () => <BlockTemplate {...defaultProps} />);
