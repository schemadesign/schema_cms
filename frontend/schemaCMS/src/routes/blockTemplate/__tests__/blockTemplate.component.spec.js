import React from 'react';

import { BlockTemplate } from '../blockTemplate.component';
import { defaultProps } from '../blockTemplate.stories';
import { makeContextRenderer } from '../../../shared/utils/testUtils';

const mockPushHistory = jest.fn();

jest.mock('react-router', () => ({
  ...jest.requireActual('react-router'),
  useHistory: () => ({
    push: mockPushHistory,
  }),
  useParams: () => ({
    blockTemplateId: 'blockTemplateId',
  }),
}));

describe('BlockTemplate: Component', () => {
  const render = props => makeContextRenderer(<BlockTemplate {...defaultProps} {...props} />);

  it('should render correctly', async () => {
    const wrapper = await render();
    expect(wrapper).toMatchSnapshot();
  });

  it('should render called', async () => {
    jest.spyOn(defaultProps, 'fetchBlockTemplate');
    await render();
    expect(defaultProps.fetchBlockTemplate).toHaveBeenCalledWith({ blockTemplateId: 'blockTemplateId' });
  });

  it('should redirect to block templates', async () => {
    const wrapper = await render();
    wrapper.root.findByProps({ id: 'cancelBtn' }).props.onClick();

    expect(mockPushHistory).toHaveBeenCalledWith('/project/projectId/block-templates');
  });
});
