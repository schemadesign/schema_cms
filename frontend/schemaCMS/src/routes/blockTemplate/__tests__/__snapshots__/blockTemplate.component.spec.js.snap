// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlockTemplate: Component should render correctly 1`] = `
<div
  className="sc-bdVaJa bDWFJH"
>
  <div
    className="sc-bwzfXH jguWiy"
  >
    <div
      style={
        Object {
          "alignItems": "flex-start",
          "color": "#FFF",
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingBottom": "18px",
          "width": "100%",
        }
      }
    >
      <div>
        <div
          className="sc-htpNat KtFsv"
        >
          <h2
            className="sc-dnqmqq bPutyk"
            id="headerTitle"
            style={
              Object {
                "fontFamily": "Inter, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
                "fontSize": "24px",
                "fontWeight": 100,
                "margin": 0,
              }
            }
          >
            Block Template
          </h2>
          <h1
            className="sc-iwsKbI bxRSJa"
            id="headerSubtitle"
            style={
              Object {
                "fontFamily": "Inter, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
                "fontSize": "24px",
                "fontWeight": 600,
                "letterSpacing": "-0.12px",
                "lineHight": 1.25,
                "margin": 0,
              }
            }
          >
            Edit
          </h1>
        </div>
      </div>
      <button
        disabled={false}
        id="topHeaderOpenMenuBtn"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "transparent",
            "border": "none",
            "borderRadius": "48px",
            "color": "#FFF",
            "cursor": "pointer",
            "display": "inline-block",
            "fill": "#FFF",
            "fontSize": "18px",
            "margin": "0",
            "minHeight": "48px",
            "outline": "none",
            "padding": "0",
            "position": "absolute",
            "right": 0,
            "top": 6,
            "transition": "background 0.25s ease-in-out, color 0.25s ease-in-out",
          }
        }
        type="button"
      >
        <svg
          style={
            Object {
              "fill": "#FFF",
              "height": "60px",
              "width": "60px",
            }
          }
          viewBox="0 0 10 60"
        >
          <path
            d="M5 37a3 3 0 110 6 3 3 0 010-6zm0-10a3 3 0 110 6 3 3 0 010-6zm0-10a3 3 0 110 6 3 3 0 010-6z"
            fillRule="evenodd"
          />
        </svg>
      </button>
    </div>
    <div
      style={
        Object {
          "backgroundColor": "#000",
          "color": "#FFF",
          "maxWidth": null,
          "minHeight": "100vh",
          "padding": "18px 20px 27px",
          "position": "absolute",
          "right": 0,
          "textAlign": "left",
          "top": 0,
          "transform": "translateX(100%)",
          "transition": "transform 400ms cubic-bezier(0.86, 0, 0.07, 1), visibility 0s linear 400ms",
          "visibility": 0,
          "width": "100%",
          "zIndex": 9999,
        }
      }
    >
      <div
        className="sc-bxivhb bUZGRr"
      >
        <div
          className="sc-htpNat KtFsv"
        >
          <h2
            className="sc-dnqmqq bPutyk"
            id="headerTitle"
            style={
              Object {
                "fontFamily": "Inter, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
                "fontSize": "24px",
                "fontWeight": 100,
                "margin": 0,
              }
            }
          >
            Block Template
          </h2>
          <h1
            className="sc-iwsKbI bxRSJa"
            id="headerSubtitle"
            style={
              Object {
                "fontFamily": "Inter, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
                "fontSize": "24px",
                "fontWeight": 600,
                "letterSpacing": "-0.12px",
                "lineHight": 1.25,
                "margin": 0,
              }
            }
          >
            Edit
          </h1>
        </div>
      </div>
      <div
        className="sc-ifAKCX YtKgc"
      >
        <div
          className="sc-EHOje jndxID"
        >
          <div
            className="sc-bZQynM dHaYym"
          >
            <a
              href="/project/"
              id="projectsNavBtn"
              onClick={[Function]}
            >
              Projects
            </a>
          </div>
          <div
            className="sc-bZQynM dHaYym"
          >
            <a
              href="/user/"
              id="usersPageNavBtn"
              onClick={[Function]}
            >
              Users
            </a>
          </div>
          <div
            className="sc-bZQynM dHaYym"
          >
            <a
              href="/project/undefined"
              id="projectDetailsNavBtn"
              onClick={[Function]}
            >
              Project details
            </a>
          </div>
          <div
            className="sc-bZQynM dHaYym"
          >
            <a
              href="/project/undefined/datasource"
              id="dataSourceNavBtn"
              onClick={[Function]}
            >
              Data Sources
            </a>
          </div>
          <div
            className="sc-bZQynM dHaYym"
          >
            <a
              href="/project/undefined/state"
              id="stateNavBtn"
              onClick={[Function]}
            >
              States
            </a>
          </div>
          <div
            className="sc-bZQynM dHaYym"
          >
            <a
              href="/project/undefined/user"
              id="usersNavBtn"
              onClick={[Function]}
            >
              Project Users
            </a>
          </div>
          <div
            className="sc-bZQynM dHaYym"
          >
            <a
              href="/project/undefined/folder"
              id="folderNavBtn"
              onClick={[Function]}
            >
              Pages
            </a>
          </div>
        </div>
        <div
          className="sc-EHOje jndxID"
        >
          <div
            className="sc-bZQynM dHaYym"
          >
            <a
              href="/settings"
              id="settingsNavBtn"
              onClick={[Function]}
            >
              Settings
            </a>
          </div>
          <div
            className="sc-bZQynM dHaYym"
          >
            <div
              onClick={[Function]}
            >
              Log out
            </div>
          </div>
        </div>
        <div
          className="sc-gzVnrw jHfMTZ"
        >
          <a
            className="sc-htoDjs dmeaZZ"
            href="/"
            id="aboutNavBtn"
            onClick={[Function]}
          >
            About Schema CMS
          </a>
          <a
            className="sc-htoDjs dmeaZZ"
            href="/"
            id="apiNavBtn"
            onClick={[Function]}
          >
            API Doc
          </a>
          <a
            className="sc-htoDjs dmeaZZ"
            href="/"
            id="repositoryNavBtn"
            onClick={[Function]}
          >
            GitHub Repository
          </a>
        </div>
      </div>
      <button
        disabled={false}
        id="topHeaderCloseMenuButton"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "transparent",
            "border": "none",
            "borderRadius": "48px",
            "color": "#FFF",
            "cursor": "pointer",
            "display": "inline-block",
            "fill": "#FFF",
            "fontSize": "18px",
            "margin": "0",
            "minHeight": "48px",
            "outline": "none",
            "padding": "0",
            "position": "absolute",
            "right": 0,
            "top": "6px",
            "transition": "background 0.25s ease-in-out, color 0.25s ease-in-out",
          }
        }
        type="button"
      >
        <svg
          style={
            Object {
              "fill": "#FFF",
              "height": "60px",
              "width": "60px",
            }
          }
          viewBox="0 0 60 60"
        >
          <path
            d="M39.192 17.98l2.829 2.828L32.828 30l9.193 9.192-2.829 2.829L30 32.828l-9.192 9.193-2.829-2.829L27.172 30l-9.193-9.192 2.829-2.829L30 27.172l9.192-9.193z"
            fillRule="evenodd"
          />
        </svg>
      </button>
    </div>
  </div>
  <form
    onSubmit={[Function]}
  >
    <div
      className="sc-kGXeez eIIwaK"
    >
      <div
        className="sc-ckVGcZ kavnbm"
      >
        <div
          className="sc-jKJlTe bbYNJN"
        >
          <h2
            className="sc-eNQAEJ jraORh"
            id="desktopHeaderTitle"
            style={
              Object {
                "fontFamily": "Inter, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
                "fontSize": "24px",
                "fontWeight": 100,
                "margin": 0,
              }
            }
          >
            Block Template
          </h2>
          <h1
            className="sc-hMqMXs NSfhp"
            id="desktopHeaderSubtitle"
            style={
              Object {
                "fontFamily": "Inter, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
                "fontSize": "24px",
                "fontWeight": 600,
                "letterSpacing": "-0.12px",
                "lineHight": 1.25,
                "margin": 0,
              }
            }
          >
            <div
              className="sc-kpOJdX dEeRhR"
            >
              <div
                className="sc-kkGfuU KklVJ"
              >
                <div
                  style={
                    Object {
                      "display": "flex",
                      "flexDirection": "column",
                      "maxWidth": "100%",
                      "paddingBottom": 0,
                      "paddingRight": 40,
                      "position": "static",
                    }
                  }
                >
                  <input
                    id="name"
                    name="name"
                    onBlur={[Function]}
                    onChange={[Function]}
                    placeholder="Name"
                    style={
                      Object {
                        "backgroundColor": "transparent",
                        "border": "none",
                        "color": "#FFF",
                        "fontFamily": "inherit",
                        "fontSize": 24,
                        "fontWeight": 600,
                        "letterSpacing": "normal",
                        "lineHeight": "24px",
                        "maxWidth": "100%",
                        "minWidth": 100,
                        "outline": "none",
                        "padding": 0,
                        "width": 105,
                      }
                    }
                    type="text"
                    value="block name"
                  />
                  <div
                    style={
                      Object {
                        "backgroundColor": "transparent",
                        "border": "none",
                        "color": "#FFF",
                        "fontFamily": "inherit",
                        "fontSize": 24,
                        "fontWeight": 600,
                        "letterSpacing": "normal",
                        "lineHeight": "24px",
                        "maxWidth": "100%",
                        "minWidth": 100,
                        "opacity": 0,
                        "outline": "none",
                        "padding": 0,
                        "position": "absolute",
                        "visibility": "hidden",
                        "whiteSpace": "pre",
                      }
                    }
                  >
                    block name
                  </div>
                  <div
                    style={
                      Object {
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      }
                    }
                  />
                </div>
              </div>
              <div
                className="sc-dxgOiQ gLysol"
              >
                <svg
                  style={
                    Object {
                      "fill": "#FFF",
                      "height": "30px",
                      "width": "30px",
                    }
                  }
                  viewBox="0 0 30 30"
                >
                  <path
                    d="M21.789 10.92a.724.724 0 000-1.022l-1.687-1.687a.724.724 0 00-1.022 0l-1.322 1.322 2.709 2.709 1.322-1.322zM9 18.291V21h2.709l7.992-7.992-2.709-2.71L9 18.292z"
                    fillRule="evenodd"
                  />
                </svg>
              </div>
            </div>
          </h1>
        </div>
        <div
          className="sc-kEYyzF gfWKdx"
        >
          <div
            className="sc-kgoBCf ifKHSM"
          >
            <button
              disabled={false}
              id="createElement"
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "#FFF",
                  "border": "none",
                  "borderRadius": "48px",
                  "color": "#000",
                  "cursor": "pointer",
                  "display": "inline-block",
                  "fill": "#FFF",
                  "fontSize": "18px",
                  "height": "60px",
                  "margin": "0",
                  "minHeight": "48px",
                  "outline": "none",
                  "padding": "0",
                  "transition": "background 0.25s ease-in-out, color 0.25s ease-in-out",
                  "width": "60px",
                }
              }
              type="button"
            >
              <svg
                style={
                  Object {
                    "fill": "#000",
                    "height": "60px",
                    "shapeRendering": "crispEdges",
                    "width": "60px",
                  }
                }
                viewBox="0 0 60 60"
              >
                <path
                  d="M32 15v13h13v4H32v13h-4V32H15v-4h13V15h4z"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      className="sc-fjdhpX jevcr"
    >
      <div
        className="sc-cSHVUG jpEzzg"
      >
        <div
          className="sc-jzJRlG dfKXm"
        >
          <button
            className="sc-kAzzGY iLakCl"
            disabled={false}
            id="cancelBtn"
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": "#1d1d20",
                "border": "none",
                "borderRadius": "48px",
                "color": "#FFF",
                "cursor": "pointer",
                "display": "inline-block",
                "fill": "#FFF",
                "fontSize": "18px",
                "margin": "0",
                "minHeight": "48px",
                "outline": "none",
                "padding": "0",
                "transition": "background 0.25s ease-in-out, color 0.25s ease-in-out",
              }
            }
            type="button"
          >
            Back
          </button>
          <button
            className="sc-kAzzGY iLakCl"
            disabled={true}
            id="createTemplateBlockMobile"
            style={
              Object {
                "backgroundColor": "#1d1d20",
                "border": "none",
                "borderRadius": "48px",
                "color": "#71737e",
                "cursor": "default",
                "display": "inline-block",
                "fill": "#FFF",
                "fontSize": "18px",
                "margin": "0",
                "minHeight": "48px",
                "outline": "none",
                "padding": "0",
                "transition": "background 0.25s ease-in-out, color 0.25s ease-in-out",
              }
            }
            type="submit"
          >
            Save
          </button>
        </div>
      </div>
    </div>
  </form>
</div>
`;
